### ISSUE TITLE
Record screen crash

### ISSUE THREAD

**Issue by DannyCork** [dirkwhoffmann/vAmiga#452]
Attempting to record the screen (Machine/Record Screen) of any adf results in a vAmiga crash
 0.9.15.1 (0.9.15.1) :(

`
Process:               vAmiga [2929]
Path:                  /Applications/myapps/vAmiga0.9.15.1.app/Contents/MacOS/vAmiga
Identifier:            dirkwhoffmann.vAmiga
Version:               0.9.15.1 (0.9.15.1)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           vAmiga [2929]
User ID:               501

Date/Time:             2020-12-08 00:59:34.809 +0000
OS Version:            macOS 11.0.1 (20B29)
Report Version:        12
Bridge OS Version:     5.0.1 (18P2561)
Anonymous UUID:        19CA2E0F-AC20-E54B-54A5-5A756C8CC505

Time Awake Since Boot: 4200 seconds

System Integrity Protection: enabled

Crashed Thread:        13

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
Assertion failed: (clock >= c1 && clock < c2), function interpolate, file /Users/hoff/vAmiga/Emulator/Paula/Audio/Sampler.cpp, line 53.

**Comment by alpine9000** [comment]
I get the same behaviour - was working previously for me

**Comment by dirkwhoffmann** [comment]
Crashes here, too. I think I already know what it is.

**Comment by dirkwhoffmann** [comment]
Found it. As part of my "Making vAmiga compatible with Linux" project, I changed this

```
   // Inputs (one Sampler for each of the four channels)
   Sampler sampler[4];
```
to that:

```
   // Inputs (one Sampler for each of the four channels)
   Sampler *sampler[4];
```

Since I made this change, the following code in the ScreenRecorder class obviously did no longer do what it is supposed to do:

 ```
      // Clone Paula's muxer contents
      muxer.sampler[0] = paula.muxer.sampler[0];
      muxer.sampler[1] = paula.muxer.sampler[1];
      muxer.sampler[2] = paula.muxer.sampler[2];
      muxer.sampler[3] = paula.muxer.sampler[3];
```

**Comment by dirkwhoffmann** [comment]
Fixed in v0.9.15.2.

Please reopen if the issue recurs.

**Comment by alpine9000** [comment]
I get a crash/exit now if I stop recording, then restart it (with a game running with sound and disk access). No crash log is generated.

**Comment by dirkwhoffmann** [comment]
> I get a crash/exit now if I stop recording, then restart it 

Thanks for reporting! With the latest checkin, it no longer crashes. 

However, I am unsure about the best usage model. The current implementation overwrites any previously recorded data when the screen recorder is startet. In other words: If no file export is made in between, only the most recently recorded part will be saved. On the one hand, this is a suitable behavior for the use-case that the user recorded a wrong or bad scene. He could simply record again and just save to disk when he thinks the scene is good enough. On the other hand, the user might want to record multiple distinct parts of a game or demo. This wouldn't work with the current implementation.

**Comment by alpine9000** [comment]
For me, I think the current behaviour is fine. If I want to keep a recording I'll export it. If I re-record without exporting it's because I have made a mistake and recorded the wrong part.

**Comment by dirkwhoffmann** [comment]
Should be fixed in v0.9.16.
